test/common/crypto/utility_test.cc

```diff
@@ -159,30 +212,816 @@ TEST(UtilityTest, TestVerifySignature) {
+      "089b003f44528aa7192591132c265083a9f99e0dca9f4039a77ab963deb0a277c168e9735124855870b02774845c"
+      "9172635e67646ec9c265868fc804c967427c87be3e3819c9539d9fb27670c85bc179de6959443492c9174a423aff"
+      "488678be35f9f003d7adeab92d7972349e5f5a4d21ecc9eecb812132dfcec4477454e09c07f51684df4720e04a9e"
+      "24362db8cd2196c1804782a682174b4dc977a84eb27c1f664f22eb64b3abae433d045fb4eea3730bc4ef30d0fb85"
+      "98471dea2c78f654ebcded8b7436155c1f03362e8409c0636022b8116bced4c46099c53fa4d8d8d1f4f6be7775fd"
+      "448ea888444da102818100f11fb88f8514202d4e3b137270f3cb98d8e17fc9caf77c76eda9a1bc0e2cebc4c3997a"
+      "bf96bcdb945beede3e01d6464913f446d594218677619ecdb584b63dca81cacd9fa9030a00d5bb143483b8aaa86a"
+      "7d8616adc16645376c8904e259e784e5fced37135ea8f776940cd3371550acdc1af2d409bfc1ad7253ab1541540f"
+      "dd02818100db3602515c160b41803d732afbbb8f411fc024648932e44e7dd8e728cbfe7bc5282a6f57027964c8ba"
+      "22618a83f1161d187251efd5de3bb7c83d50db6295b1392e9e87c205761858daed057317d815cafe52253eaf2f72"
+      "6897965ed46f0a212d8355a2d2e64882e9e32166cca7e4336cc3b279ace0f67abee126e39087682e8302818100ec"
+      "091b481303a283f722c964abc15bba62044c6da32c2540de61c19b2f5d35e6c57ac6b829bcf24e06b88c01b316a8"
+      "72fcff911f9e043b773dae90bc720f5be992a88e250ef394a5409403b16c882736fa17aa5d24f63f40de827696bb"
+      "653ac7d3c3860af60121f22cb7bcde3dfbb59fa14f180a0d091374d087aae001b5625902818011561922d4148e39"
+      "54ea0734ac09ee4f693269ee658757d4f950f11f21daf370e93749ece8ae2f114cdf3135a22fabdf0b32e755ff64"
+      "fef60ee9027f0731ed7d2739b464dcc7b52f39c92af82a3795a9a3295df6b2261f77341dd94c15a8086db00852c3"
+      "39211cf1605c20e42896fc962a77eff583291b16037a6ededc4699ff02818100cadc0cbd4e4f00301e3594190529"
+      "c8324c19ed77138b7582288a229f86c6f261f95b93d47a318856b3585e68b1b90be6c8467a4e8f97f6e820064f8d"
+      "2793ddf93e1cfa119f1f166de15d6588d9e8ac5ffd30c953374c22557d3f80d24982425dfe00754cfab810c8ff12"
+      "6adfb09964d360d1d2d337cf3076c53e4d59f911feee";
+
+  Common::Crypto::PKeyObjectPtr crypto_ptr(
+      Common::Crypto::UtilitySingleton::get().importPrivateKeyDER(Hex::decode(private_key)));
+  Common::Crypto::PKeyObject* crypto(crypto_ptr.get());
+
+  // Test signing with extremely large text to potentially trigger EVP_DigestSign errors
+  std::vector<uint8_t> huge_text(100000000, 'X'); // 100MB of data
+  auto result = UtilitySingleton::get().sign("sha256", *crypto, huge_text);
+  // This should still succeed but exercises the signature length/creation paths
+  ASSERT_TRUE(result.ok()) << "Large text signing should succeed: " << result.status();
+  EXPECT_FALSE(result->empty()) << "Large text signature should not be empty";
+
+  // Test with corrupted key that might cause EVP_DigestVerifyInit to fail
+  auto corrupted_key = std::make_unique<PKeyObject>();
+  // Create a partially valid but corrupted EVP_PKEY that might cause init failures
+  EVP_PKEY* corrupted_pkey = EVP_PKEY_new();
```

Comment:
How is this corrupt? I'm not really sure what you're trying to validate with this test.